pkg/configprovider/providerutil/utils_test.go

```diff
@@ -0,0 +1,91 @@
+package providerutil
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/confmap"
+)
+
+func TestPrepareForReplaceBehavior(t *testing.T) {
+	tests := []struct {
+		name         string
+		srcConf      *confmap.Conf
+		mergeConf    *confmap.Conf
+		expectedConf *confmap.Conf
+	}{
+		{
+			name: "Remove matching key path from source map",
+			srcConf: confmap.NewFromStringMap(map[string]interface{}{
+				"extensions": map[string]interface{}{
+					"sumologic": map[string]interface{}{
+						"childKey2": "value2",
+						"collector_fields": map[string]interface{}{
+							"cluster": "cluster-1",
+						},
+					},
+				},
+				"anotherKey": "anotherValue",
+			}),
+			mergeConf: confmap.NewFromStringMap(map[string]interface{}{
+				"extensions": map[string]interface{}{
+					"sumologic": map[string]interface{}{
+						"collector_fields": map[string]interface{}{
+							"zone": "eu",
+						},
+						"childKey2": "value2",
+					},
+				},
+				"anotherKey": "anotherValue",
+			}),
+			expectedConf: confmap.NewFromStringMap(map[string]interface{}{
+				"extensions": map[string]interface{}{
+					"sumologic": map[string]interface{}{
+						"childKey2": "value2",
```

Comment:
why isn't "zone": "eu" the  expected field here?

Reply:
Have added just some sample test values. Do you think we should have specific values here?

Reply:
I mean, I don't understand the assertion here. My questions are
- why doesn't the expected config have any collector_fields.
- why are we removing `"cluster": "cluster-1"` from the src/original conf? and why isn't `collector_fields` from `mergeConf` in the expected result?

Reply:
@jagan2221 would appreciate clarification here, maybe I'm missing something?

Reply:
agreed with Raj, the expected config should have something in ```collector_fields```. i remember there was a sumo documentation showing the merge functionality, think it was this: 

https://help.sumologic.com/docs/send-data/opentelemetry-collector/data-source-configurations/overview/#configuration-structure:~:text=Note%20that%20the%20list%20has%20been%20overridden%20and%20maps%20have%20been%20merged.

Reply:
@jagan2221 shouldn't the `collector_fields` in the expected config have `"zone": "eu"` since we now replace the existing tag instead of appending to it? Also, am I right in understanding that `mergeConf` is simply a second configuration file?

I have listed the expected config below

Original behavior

Expected config
```
collector_fields:
  "cluster": "cluster-1"
  "zone": "eu"
```

New (Replace) behavior

Expected config
```
collector_fields:
  "zone": "eu"
```

what am I missing here?

Reply:
oh after staring at this for a bit, think I know why the expected config is this:
```
expectedConf: confmap.NewFromStringMap(map[string]interface{}{
    "extensions": map[string]interface{}{
	    "sumologic": map[string]interface{}{
		    "childKey2": "value2",
	    },
    },
    "anotherKey": "anotherValue",
}),
```

since Jagan mentioned:

> We are not testing the merging part, The func PrepareForReplaceBehaviour prepares sourceConf for replace behaviour, ie) If sourceConf has collector_fields key and merge conf also has the same , then collector_fields key will be removed from sourceConf.

> Actual merge flow remains the same, we are just overriding sourceConf(by removing collector_fields key based on condition that mergeConf also has it), so that existing merge will add collector_fields from mergeConf into expectedConf

this is testing ONLY the replacing functionality and not the merging. if so, then this test makes sense. what made it confusing for me was because I'm assuming this would also test for the merging functionality. 

the merging logic happens in [globprovider](https://github.com/SumoLogic/sumologic-otel-collector/blob/3e13dd3d4dd3a3b88ebd22b6c8988452123203fb/pkg/configprovider/globprovider/provider.go#L71-L72)

now i'm wondering should a new test be added in [globprovider test](https://github.com/SumoLogic/sumologic-otel-collector/blob/3e13dd3d4dd3a3b88ebd22b6c8988452123203fb/pkg/configprovider/globprovider/provider_test.go) that will specifically test the whole flow of replacing to merging flow.

Reply:
I see, I wonder why none of the existing tests failed, with this change in merge behavior. But I tend to agree, I think we need a test for the whole flow including the merge. I'd rather be too careful for a feature that potentially impacts all of our configuration than otherwise. Thanks for being patient.

Reply:
Added test case for whole merge flow with enable/disable new merge behaviour in globprovider test
@rnishtala-sumo @chan-tim-sumo